src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/FhirRequestContextDocumentClientExtensions.cs

```diff
@@ -0,0 +1,77 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Globalization;
+using System.Net;
+using EnsureThat;
+using Microsoft.Azure.Documents;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Health.Fhir.Core.Exceptions;
+using Microsoft.Health.Fhir.Core.Features.Context;
+
+namespace Microsoft.Health.Fhir.CosmosDb.Features.Storage
+{
+    internal static class FhirRequestContextDocumentClientExtensions
+    {
+        /// <summary>
+        /// Adds request charge to the response headers and throws a <see cref="RequestRateExceededException"/>
+        /// if the status code is 429.
+        /// </summary>
+        /// <param name="requestContext">The request context</param>
+        /// <param name="ex">The exception</param>
+        public static void ProcessException(this IFhirRequestContext requestContext, Exception ex)
+        {
+            if (requestContext == null)
```

Comment:
should this be EnsureArg.IsNotNull(requestContext, nameof(requestContext))? It works but semantically calling requestContext.ProcessException where requestContext is null is a little weird :)

Reply:
Taking this approach instead of having to do null checks at all the call sites. Agree it is a bit weird.